tensorflow/contrib/layers/kernels/layer_norm_fused_op_gpu.cu.cc

```diff
@@ -0,0 +1,1455 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include <math.h>
+#include <stdio.h>
+#include <algorithm>
+
+#if GOOGLE_CUDA
+#define EIGEN_USE_GPU
+
+#include "./layer_norm_fused_op.h"
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+
+#if !defined(_MSC_VER)
+#define UNROLL _Pragma("unroll")
+#else
+#define UNROLL
+#endif
+
+#if !defined(__CUDA_ARCH__) || __CUDA_ARCH__ >= 600
+#else
+__device__ double atomicAdd(double* a, double b) { return b; }
+#endif
+
+#define MAX_GRID_SIZE 480
+inline int get_num_blocks(const int n_slices, const int slice_per_block) {
+  const int _num_blocks = n_slices / slice_per_block;
+  if (_num_blocks * slice_per_block == n_slices)
+    return _num_blocks;
+  else
+    return _num_blocks + 1;
+}
+
+inline int get_block_size(const int slice_size, int& block_size, int& mult) {
+  const int _warp_size = 32;
+  int _block_size = _warp_size;
+  int _mult = slice_size / _block_size;
+  mult = _mult * _block_size >= slice_size ? _mult : _mult + 1;
+  while (mult > 5) {
+    _block_size += _warp_size;
+    _mult = slice_size / _block_size;
+    mult = _mult * _block_size >= slice_size ? _mult : _mult + 1;
+  }
```

Comment:
Yes... I stuck on this part for a long time, but eventually compromised on this solution, because most of my use cases had slice_size <= 1500, so the loop will not go beyond 10 steps. Also, I thought the  extra time needed for this loop would be many times shorter than the kernel itself, so I was a bit reluctant on optimizing this part.